docs/cookbooks/creating_a_derived_dataset.md

```diff
@@ -0,0 +1,208 @@
+
+You'll also note the `public_json` and `public_bigquery` fields: these mean that the dataset will be published in both Mozilla's public BigQuery project and a world-accessible JSON endpoint. The `review_bugs` section is required, and refers to the Bugzilla bug where opening this data set up to the public was approved: we'll get to that in a subsequent section.
+
+Now that we've filled out the metadata, we can look into creating a query. In many ways, this is similar to creating a SQL query to run on BigQuery in other contexts (e.g. on sql.telemetry.mozilla.org or the BigQuery console)-- the key difference is that we use a `@submission_date` parameter so that the query can be run on a _day's worth_ of data to update the underlying table incrementally.
+
+After testing our query on sql.telemetry.mozilla.org, we'll write this query out to `query.sql`:
+
+```sql
+SELECT
+  DATE(submission_timestamp) AS date,
+  client_info.app_display_version AS mozregression_version,
+  metrics.string.usage_variant AS mozregression_variant,
+  metrics.string.usage_app AS app_used,
+  normalized_os AS os,
+  mozfun.norm.truncate_version(normalized_os_version, "minor") AS os_version,
+  count(DISTINCT(client_info.client_id)) AS distinct_clients,
+  count(*) AS total_uses
+FROM
+  `moz-fx-data-shared-prod`.org_mozilla_mozregression.usage
+WHERE
+  DATE(submission_timestamp) = @submission_date
+  AND client_info.app_display_version NOT LIKE '%.dev%'
+GROUP BY
+  date,
+  mozregression_version,
+  mozregression_variant,
+  app_used,
+  os,
+  os_version;
+```
+
+We use the `truncate_version` UDF to omit the patch level for MacOS and Linux, which should both reduce the size of the dataset as well as make it more difficult to identify individual clients in an aggregated dataset.
+
+We also have a short clause (`client_info.app_display_version NOT LIKE '%.dev%'`) to omit developer versions from the aggregates: this makes sure we're not including people developing or testing mozregression itself in our results.
+
+## Formatting and validating the query
+
+Now that we've written our query, we can format it and validate it. Once that's done, we run:
+
+```bash
+./bqetl query validate org_mozilla_mozregression_derived.mozregression_aggregates
+```
+
+If there are no problems, you should see no output.
+
+## Creating a DAG
+
+BigQuery-ETL has some facilities in it to automatically add your query to [telemetry-airflow](https://github.com/mozilla/telemetry-airflow) (our instance of Airflow).
+
+Before scheduling your query, you'll need to find an [Airflow DAG](https://airflow.apache.org/docs/apache-airflow/stable/concepts.html#dags) to run it off of. In some cases, one may already exist but in this one, we don't have one -- so we'll create it:
```

Comment:
Is it worth mentioning here how to find existing DAGs? Either by looking into the `dags.yaml` or by running `bqetl dag info`

Reply:
Yup! Added some notes about this.